api/forms.py

```diff
@@ -0,0 +1,94 @@
+# encoding: utf-8
+from django import forms
+from django.forms.widgets import (TextInput, Textarea, RadioSelect, CheckboxSelectMultiple,
+                                 MultiWidget, Select, SelectMultiple, DateInput, DateTimeInput,
+                                 FileInput, NumberInput, EmailInput, URLInput, PasswordInput,
+                                 HiddenInput)
+from crispy_forms.helper import FormHelper
+from crispy_forms.utils import render_crispy_form
+
+
+class HopperForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        model = kwargs.pop('model', None)
+        elements = model.convert_to_dict(model.elements)
+        super(HopperForm, self).__init__(*args, **kwargs)
+        self.type_widget_mapping = {
+            'input': TextInput,
+            'fieldset': MultiWidget,
+            'textarea': Textarea,
+            'radio': RadioSelect,
+            'checkbox': CheckboxSelectMultiple,
+            'select': Select,
+            'multiselect': SelectMultiple,
+            'date': DateInput,
+            'datetime': DateTimeInput,
+            'file': FileInput,
+            'integer': NumberInput,
+            'mail': EmailInput,
+            'url': URLInput,
+            'password': PasswordInput,
+            'hidden': HiddenInput,
+        }
+        self.helper = FormHelper()
+        self.helper.form_class = model.css_classes
+        self.helper.form_action = model.action
+        self.helper.form_method = model.method
+        self.helper.field_class = model.elements_css_classes
+        self.helper.form_show_labels = False
+        self.fields = self.create_fields(elements)
+
+    def create_fields(self, elements):
+        """Creates dictionary with fields and its attributes and
+        widgets"""
+        fields = {}
+        for name, element in elements.items():
+            field_attrs = self.get_base_field_attrs(element)
+            fields[name] = forms.Field(field_attrs, widget=self.create_widget(element))
+            if element['type'] == 'fieldset':
+                fields.update(self.create_fields(element['elements']))
```

Comment:
Currently you have no layout defined. Doing it this way, you will have not FieldSets in the rendered template.